app/models/tag.rb

```diff
@@ -8,6 +8,20 @@ class Tag < ApplicationRecord
 
   before_destroy :remove_from_managed_filters
 
+  def self.list(taggable, options = {})
+    ns = get_namespace(options)
+    if ns[0..7] == "/virtual"
+      predicate = ns.split("/")[2..-1] # throw away /virtual
+      begin
+        predicate.inject(taggable) { |target, method| target.public_send method }
+      rescue NoMethodError
+        ""
+      end
+    else
+      filter_ns(taggable.tags, ns).join(" ")
```

Comment:
Reviewing this with @imtayadeway we realized that this could still blow up if call with a `taggable` whose class is not taggable.